src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -303,9 +303,12 @@ CHIP_ERROR DeviceControllerFactory::SetupController(SetupParams params, DeviceCo
 CHIP_ERROR DeviceControllerFactory::SetupCommissioner(SetupParams params, DeviceCommissioner & commissioner)
 {
     VerifyOrReturnError(mSystemState != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(params.controllerVendorId != VendorId::Unspecified, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Should put this check in `DeviceControllerFactory::SetupController` too, right?

Reply:
I didn't put it there since controllerVendorId only seems to be necessary for commissioners?

Reply:
Was trying to minimize the required set of bits controllers vs. commissioners needed to specify to be functional.

Reply:
It's used in two places:

1. To set `mVendorId`, which seems to only be used in `DeviceCommissioner::SendOperationalCertificate`.
2. To set up the fabric table entry, which then might get used by anyone.

Reply:
Fair enough. Will update `SetupController` too.